### PR TITLE
ToffeeIR update.

### PR DESCRIPTION
**Pull request by ezyang**
- kernels -> kernel_shape
- Use the new hybrid dict/tuple result object from Toffee
- Write g and t as singulars, not plural
- nanopb generated files update
- Bugfix for msg() micropb helper
- Start recording producer_version/producer_tag
- Use ir_version from proto description
- Value -> value (Constant)
- Remove special-casing for transposed convolution; we now rely
  on the Caffe2 Toffee backend to do something reasonable
- Batchnorm order is no more

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/csrc/toffee/export.cpp [hunk 1]
```diff
@@ -233,8 +233,7 @@ static void addAttribute(toffee::NodeProto * n_p, jit::Node * n, jit::Symbol nam
         attr->add_strings(v);
       break;
     case AttributeKind::t: {
-      //TODO: tensors but no tensor?
-      auto t = attr->add_tensors();
+      auto t = attr->mutable_t();
       encodeTensor(t, n->t(name));
```
Comment (zdevito) [inline torch/csrc/toffee/export.cpp:237]: Hurray! This inconsistency survived for a really long time :)

**Review by zdevito — APPROVED** [review]
Looks good.

File: torch/csrc/autograd/functions/toffee/batch_normalization.cpp [hunk 2]
```diff
@@ -14,7 +14,7 @@ jit::node_list BatchNormForward::primspec(PrimSpecContext* ctx, jit::node_list i
   bn->addInput(jit::tracer::getBufferTrace(*ctx->buffer_map, cached_running_var));
   bn->i_(jit::kis_test, !this->training);
   bn->f_(jit::kepsilon, eps);
-  bn->s_(jit::korder, "NCHW");
+  //bn->s_(jit::korder, "NCHW");
```
Comment (apaszke) [inline torch/csrc/autograd/functions/toffee/batch_normalization.cpp:17]: Why not remove it?

**Review by apaszke — APPROVED** [review]

File: torch/csrc/autograd/functions/toffee/batch_normalization.cpp (hunk 2 above)
Comment (ezyang) [reply torch/csrc/autograd/functions/toffee/batch_normalization.cpp:17]: I suppose, it is a reminder that our format is NCHW :)
